power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/dataset.hpp

```diff
@@ -88,9 +88,13 @@ class DatasetWritable {
 
 class DatasetMutable {
   public:
-    DatasetMutable(std::string const& dataset, Idx is_batch, Idx batch_size)
-        : dataset_{handle_.call_with(PGM_create_dataset_mutable, dataset.c_str(), is_batch, batch_size)},
+    DatasetMutable(std::string const& dataset, std::same_as<Idx> auto is_batch, Idx batch_size)
+        requires(!std::same_as<decltype(is_batch), bool>)
```

Comment:
yeah let me tell you a story of a man who had a stroke:
* once upon a time, there was `DatasetMutable{"input", 0, 1}` everywhere,
  * so the man tried replacing it with `DatasetMutable{"input", false, 1}`.
* but then `clang-tidy` started complaining about implicit conversions,
  * so the man added an `explicit` overload for `bool is_batch`.
* but `MSVC` did not like that, because it did not know whether `true` was a `bool` or an `Idx`,
  * so the man ended up making the `Idx` constructor `explicit`, as well.
* however, there was great evil around, because, now, `gcc` thought an `0` was `Idx` nor `bool`,
  * so the now poor man had no other choice: he extended the `Idx` constructor to all `std::convertible_to<Idx>`
* but now, all `float`s were invited as well
  * so the man made them `std::integral` in an attempt to silence now `clang-cl`
* but, and here is the catch, `bool` is also `std::integral`
  * so the man, finally giving up, decided on the current implementation

I died 3 deaths trying to solve this minor issue

Reply:
Now the lesson's been learned. But it is such a cool implementation for such a small issue. Maybe something cool to share in the tech review.